expression/builtin_time_test.go

```diff
@@ -701,3 +701,45 @@ func (s *testEvaluatorSuite) TestDateArithFuncs(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(v.IsNull(), IsTrue)
 }
+
+func (s *testEvaluatorSuite) TestTimestamp(c *C) {
+	tests := []struct {
+		t      []types.Datum
+		expect string
+	}{
+		{[]types.Datum{types.NewStringDatum("2017-01-18")}, "2017-01-18 00:00:00"},
+		{[]types.Datum{types.NewStringDatum("20170118")}, "2017-01-18 00:00:00"},
+		{[]types.Datum{types.NewStringDatum("170118")}, "2017-01-18 00:00:00"},
+		{[]types.Datum{types.NewStringDatum("20170118123056")}, "2017-01-18 12:30:56"},
+		{[]types.Datum{types.NewStringDatum("2017-01-18 12:30:56")}, "2017-01-18 12:30:56"},
+		{[]types.Datum{types.NewIntDatum(170118)}, "2017-01-18 00:00:00"},
+		{[]types.Datum{types.NewFloat64Datum(20170118)}, "2017-01-18 00:00:00"},
+		{[]types.Datum{types.NewStringDatum("20170118123050.999")}, "2017-01-18 12:30:50.999"},
+
+		{[]types.Datum{types.NewStringDatum("2017-01-18"), types.NewStringDatum("12:30:59")}, "2017-01-18 12:30:59"},
+		{[]types.Datum{types.NewStringDatum("2017-01-18"), types.NewStringDatum("12:30:59")}, "2017-01-18 12:30:59"},
+		{[]types.Datum{types.NewStringDatum("2017-01-18 01:01:01"), types.NewStringDatum("12:30:50")}, "2017-01-18 13:31:51"},
+		{[]types.Datum{types.NewStringDatum("2017-01-18 01:01:01"), types.NewStringDatum("838:59:59")}, "2017-02-22 00:01:00"},
+
+		// TODO: the following test cases exists precision questions.
+		//{[]types.Datum{types.NewFloat64Datum(20170118123950.123)}, "2017-01-18 12:30:50.123"},
+		//{[]types.Datum{types.NewFloat64Datum(20170118123950.999)}, "2017-01-18 12:30:50.699"},
+		//{[]types.Datum{types.NewFloat32Datum(float32(20170118123950.999))}, "2017-01-18 12:30:50.699"},
+
+		// TODO: the following test cases will cause time format error.
+		//{[]types.Datum{types.NewFloat64Datum(20170118.999)}, "2017-01-18 00:00:00.000"},
```

Comment:
Now, we only support the `YYYYMMDDHHMMSS.fraction` and `YYMMDDHHMMSS.fraction` format. So 20170118.999 cannot be handled now.